### ISSUE TITLE
Object Detection API - Feature Request: RetinaNet with Focal Loss

### ISSUE THREAD

**Issue by apacha** [tensorflow/models#4091]
I found some hints that indicate, that RetinaNet + Focal Loss might already be implemented in the Object Detection API, but I couldn't find any official documentation on how to use it in order to achieve the same that is done with this keras implementation: https://github.com/fizyr/keras-retinanet

Is it already implemented and ready to be used, but just undocumented? If so, it would be great to have this documented somewhere.

**Comment by xiaoyongzhu** [comment]
I am also interested in proposing this feature request. @apacha  I agree that some hints suggest someone has been working on that for a while, especially there is a whole folder talking about RetinaNet under TPU folder (https://github.com/tensorflow/tpu/tree/master/models/official/retinanet) and looks like the TPU version is already running. I suspect the code can be run using regular GPU as well.

**Comment by austinmw** [comment]
Hi, I see that there's a support message for it added for July 13th, but don't see the config file in the model zoo. Is it ran differently than the other models?

**Comment by daavoo** [comment]
@austinmw The config file is 
https://github.com/tensorflow/models/blob/master/research/object_detection/samples/configs/ssd_resnet50_v1_fpn_shared_box_predictor_640x640_coco14_sync.config

And the name in the model zoo:
ssd_resnet_50_fpn_coco ☆

**Comment by pkulzc** [comment]
That's right, it now available in model zoo. Closing this request.

**Comment by tr-ips-bahubali** [comment]
@pkulzc is there any ipynb file which has a sample code to use the ssd_resnet_50_fpn_coco?

**Comment by pkulzc** [comment]
All models in model zoo (except for tflite models) can be run in the [notebook](https://github.com/tensorflow/models/blob/master/research/object_detection/object_detection_tutorial.ipynb).

**Comment by tr-ips-bahubali** [comment]
Thanks !!

**Comment by zubairahmed-ai** [comment]
> @austinmw The config file is
> https://github.com/tensorflow/models/blob/master/research/object_detection/samples/configs/ssd_resnet50_v1_fpn_shared_box_predictor_640x640_coco14_sync.config
> 
> And the name in the model zoo:
> ssd_resnet_50_fpn_coco ☆

So this `ssd_resnet_50_fpn_coco ` is actually a RetinaNet model with ResNet base trained on coco dataset?
Why isn't it named like so?

**Comment by austinmw** [comment]
@zubairahmed-ai  I believe because RetinaNet is just SSD with a different loss function

**Comment by zubairahmed-ai** [comment]
> @zubairahmed-ai I believe because RetinaNet is just SSD with a different loss function

Is that so, I dont know better, so I can just use this config instead? https://github.com/tensorflow/models/blob/master/research/object_detection/samples/configs/ssd_resnet50_v1_fpn_shared_box_predictor_640x640_coco14_sync.config

**Comment by austinmw** [comment]
@zubairahmed-ai Yeah 

> # SSD with Resnet 50 v1 FPN feature extractor, shared box predictor and focal
> # loss (a.k.a Retinanet).

**Comment by zubairahmed-ai** [comment]
> @zubairahmed-ai Yeah
> 
> ```
> # SSD with Resnet 50 v1 FPN feature extractor, shared box predictor and focal
> # loss (a.k.a Retinanet).
> ```

Yes, yes I got it buddy, I will train my model with this now. Thanks a bunch

**Comment by nathantsoi** [comment]
Hi all, I tried to train this model (at b3158fb0183809400e9e7f8092dd541201b1c4d4) using the trainval35k coco split, fine tuning from http://download.tensorflow.org/models/resnet_v1_50_2016_08_28.tar.gz

![screen shot 2019-02-04 at 9 02 24 am](https://user-images.githubusercontent.com/399770/52223954-b37d9e80-285b-11e9-89c7-e9771d1f4f42.png)
![screen shot 2019-02-04 at 9 02 14 am](https://user-images.githubusercontent.com/399770/52223955-b37d9e80-285b-11e9-95c5-48bd1bbde7f6.png)

However, it seems like there is a problem with classification. Can anyone else confirm this result?
